modules/dxp/apps/search-experiences/search-experiences-test/src/testIntegration/java/com/liferay/search/experiences/internal/blueprint/test/SXPBlueprintSearchResultTest.java

```diff
@@ -959,6 +961,56 @@ public void testHideByExactTermMatch() throws Exception {
 		_assertSearchIgnoreRelevance("[In-Folder, Out of the folder]");
 	}
 
+	@Test
+	public void testHideComments() throws Exception {
+		JournalArticle article = JournalTestUtil.addArticle(
+			_group.getGroupId(), 0,
+			PortalUtil.getClassNameId(JournalArticle.class),
+			HashMapBuilder.put(
+				LocaleUtil.US, "Article"
+			).build(),
+			null,
+			HashMapBuilder.put(
+				LocaleUtil.US, StringPool.BLANK
+			).build(),
+			LocaleUtil.getSiteDefault(), false, true, _serviceContext);
+
+		_journalArticles.add(article);
+
+		CommentManagerUtil.addComment(
+			_user.getUserId(), _serviceContext.getScopeGroupId(),
+			JournalArticle.class.getName(), article.getResourcePrimKey(),
+			"Comment", new IdentityServiceContextFunction(_serviceContext));
+
+		_sxpBlueprint.setConfigurationJSON(
+			JSONUtil.put(
+				"generalConfiguration",
+				JSONUtil.put(
+					"searchableAssetTypes",
+					JSONUtil.put("com.liferay.message.boards.model.MBMessage"))
+			).put(
+				"queryConfiguration", JSONUtil.put("applyIndexerClauses", false)
+			).toString());
+
+		_updateSXPBlueprint();
+
+		_updateElementInstancesJSON(
+			new Object[] {_getDefaultValuesForTextMatchOverMultipleFields()},
+			new String[] {"Text Match Over Multiple Fields"});
+
+		_keywords = "Comment";
+
+		_assertSearch("[Comment]");
+
+		_updateElementInstancesJSON(
+			new Object[] {
+				_getDefaultValuesForTextMatchOverMultipleFields(), null
+			},
+			new String[] {"Text Match Over Multiple Fields", "Hide Comments"});
+
+		_assertSearch("[]");
```

Comment:
Here and a few other places, we want to make sure that the filtering doesn't just remove results but also still allows some through. Add a JournalArticle or something like this to this test (and add that type to the searchable asset types) that we would still expect to be returned in this last case.